pkg/karmadactl/cmdinit/kubernetes/node.go

```diff
@@ -26,43 +26,34 @@ func (i *CommandInitOption) getKarmadaAPIServerIP() error {
 		return nil
 	}
 
-	nodeClient := i.KubeClientSet.CoreV1().Nodes()
-	masterNodes, err := nodeClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master"})
+	nodes, err := i.KubeClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
-	if len(masterNodes.Items) == 0 {
-		masterNodes, err = nodeClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/control-plane"})
-		if err != nil {
-			return err
+
+	hasMaster := false
+	for ni := range nodes.Items {
+		if isMasterNode(&nodes.Items[ni]) {
+			hasMaster = true
+			i.KarmadaAPIServerIP = append(i.KarmadaAPIServerIP, utils.StringToNetIP(nodes.Items[ni].Status.Addresses[0].Address))
 		}
 	}
 
-	if len(masterNodes.Items) == 0 {
-		klog.Warning("the kubernetes cluster does not have a Master role.")
-	} else {
-		for _, v := range masterNodes.Items {
-			i.KarmadaAPIServerIP = append(i.KarmadaAPIServerIP, utils.StringToNetIP(v.Status.Addresses[0].Address))
-		}
+	if hasMaster {
 		return nil
 	}
 
+	klog.Warning("the kubernetes cluster does not have a Master role.")
 	klog.Info("randomly select 3 Node IPs in the kubernetes cluster.")
-	nodes, err := nodeClient.List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-
 	for number := 0; number < 3; number++ {
 		if number >= len(nodes.Items) {
-			break
+			if len(i.KarmadaAPIServerIP) == 0 {
```

Comment:
This code seems not necessary to embed here. Putting it the end could be clearer.

Reply:
I apologize for not replying to your message for such a long time (I am infected with COVID-19 and have just improved recently😀)

I adjusted the original code logic

Reply:
:) I'm infected with COVID-19 as well...